src/compilation/config.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    content_encoding::{
+        ContentTypeCheckType, ContentTypeConverterType, DEFAULT_CONTENT_ENCODING_CHECK,
+        DEFAULT_CONTENT_ENCODING_CONVERT,
+    },
+    media_type::{MediaTypeCheckType, DEFAULT_MEDIA_TYPE_CHECK},
+    schemas,
+};
+use serde_json::Value;
+use std::{collections::HashMap, fmt};
+
+#[allow(missing_docs)]
+#[derive(Clone, Default)]
+pub struct JSONSchemaCompileConfig {
+    draft: Option<schemas::Draft>,
+    custom_content_media_type_checks: HashMap<String, Option<MediaTypeCheckType>>,
```

Comment:
I think `custom_` prefix is not necessary

Reply:
I agree ... as there was no doc I preferred the redundancy for now so at least it passed the message of what we should store in the field 😎